shopfloor_base_default_profile/models/shopfloor_app_user_profile.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 Brian McMaster <brian@mcmpest.com>
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+from odoo import fields, models
+
+
+class ShopfloorAppUserProfile(models.Model):
+    _name = "shopfloor.app.user.profile"
+
+    user_id = fields.Many2one("res.users", string="User")
+    shopfloor_app_id = fields.Many2one("shopfloor.app", string="Shopfloor App")
```

Comment:
You must have a constraint here: you can have only one default profile per app per user.